products/batch_exports/backend/temporal/destinations/s3_batch_export.py

```diff
@@ -236,469 +217,6 @@ def __init__(self, message: str = "Endpoint URL is invalid."):
         super().__init__(message)
 
 
-Part = dict[str, str | int]
-
-
-class S3MultiPartUploadState(typing.NamedTuple):
-    upload_id: str
-    parts: list[Part]
-
-
-class S3MultiPartUpload:
-    """An S3 multi-part upload.
```

Comment:
I was actually thinking of keeping this, as I think that the consumer shouldn't be maintaining multi part upload state, but instead be composed with this to actually handle state.

Reply:
I rather operate on the code as it is right now rather than my future plans for it. And removing this duplication now is overall a net positive.